Tasks/Src/RTOS_BPS.c

```diff
@@ -4,17 +4,27 @@ RTOS_BPS library includes all the wrapper functions for a functioning RTOS - nee
 
 #include "RTOS_BPS.h"
 
-void RTOS_BPS_SemPend(void){ 
+void RTOS_BPS_SemPend(void) {
 
 } 
-void RTOS_BPS_SemPost(void){
+void RTOS_BPS_SemPost(void) {
 
 }
 void RTOS_BPS_MutexPend(void){
 
 }
-void RTOS_BPS_MutexPost(void){
 
+/**
+ * @brief   Posts the specified Mutex. (For future reference, Post is the same as Give)
+ * @param   *mutex - pointer to the specified RTOS Mutex object
+ * @param   options - a parameter which determines what kind of Post MutexPost performs
+ * @return  BPS_OS_ERR value
+ */
+BPS_OS_ERR RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options) {
+    BPS_OS_ERR err;
+    OSMutexPost(mutex, options, &err);
+    assertOSError(err);
+    return err;
```

Comment:
I know you probably did this because I told you to, but after a second look. `assertOSError()` already does the error handling so we don't really need to return error.

Reply:
Should I modify the function and documentation code to reflect this

Reply:
yes